scripts/travis/setup-docker-buildx.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+set -eux
+
+# update docker
+# for setup buildx https://travis-ci.community/t/installing-docker-19-03/8077/2
+sudo apt update
+
+sudo systemctl stop docker
+sudo apt install -y docker.io
+
+sudo systemctl unmask docker.service
+sudo systemctl unmask docker.socket
+sudo systemctl start docker
+sudo systemctl status docker.socket
+
+docker version
+
+DOCKER_BUILDX_VERSION=v0.4.2
+
+LOCAL_OS=$(uname -s | tr '[A-Z]' '[a-z]')
+
+case $(uname -m) in
+x86_64)
+  LOCAL_ARCH=amd64
+  ;;
+aarch64)
+  LOCAL_ARCH=arm64
+  ;;
+*)
+  echo "unsupported architecture"
+  exit 1
+  ;;
+esac
+
+if [[ ! -f ~/.docker/cli-plugins/docker-buildx ]]; then
+  DOCKER_BUILDX_DOWNLOAD_URL=https://github.com/docker/buildx/releases/download/${DOCKER_BUILDX_VERSION}/buildx-${DOCKER_BUILDX_VERSION}.${LOCAL_OS}-${LOCAL_ARCH}
+  mkdir -p ~/.docker/cli-plugins
+  echo "downloading from ${DOCKER_BUILDX_DOWNLOAD_URL}"
+  curl -sL --output ~/.docker/cli-plugins/docker-buildx "${DOCKER_BUILDX_DOWNLOAD_URL}"
+  chmod a+x ~/.docker/cli-plugins/docker-buildx
+fi
+
+# enable buildx
+export DOCKER_CLI_EXPERIMENTAL=enabled
+
+# checkout buildx available
+docker buildx version
+
+# enabled qemu if needed
+if [[ ! $(docker buildx inspect default | grep Platforms) == *arm64* ]]; then
+  docker run --privileged --rm tonistiigi/binfmt --install all
```

Comment:
I noticed you're using `tonistiigi/binfmt` image. I guess this is because it has images for all architecture which we need. Can we use append the tag too `tonistiigi/binfmt:<tag>` Right now it is pulling latest tag and the owner updating the latest image may break Jaeger travis CI.

Reply:
`tonistiigi/binfmt` is not provide the images we need for architecture.

it just install `qemu` as simulator for target arch. 
so we could run arm64 binary on amd64 host. (but it is a very slow way, and we don't use this for compiling)

the images `alpine:3.12`, you could see the arch list in docker hub.

![image](https://user-images.githubusercontent.com/1667873/97262364-a092d500-185b-11eb-9f45-ccd19bd98c0e.png)

This means docker hub store each image for each arch.

When we use `FROM alpine:3.12`, it will pull the matched arch image to local. 

like docker with `--platform` did.
```
docker pull --platform=linux/arm64 alpine:3.12
docker pull --platform=linux/amd64 alpine:3.12
docker pull --platform=linux/s390x alpine:3.12
```

With `buildx`,  the  `FROM alpine:3.12` in Dockerfile, pulled for each arch image too.
And in `RUN apt add git`, `apt` (arm64 binary) need to execute on matched arch (arm64) host.
This why we need `qemu`. with `qemu` we could run `apt` (arm64 binary) on arm64 host. 

And we don't need the tag.  it is a official tool in [buildx doc](https://github.com/docker/buildx#building-multi-platform-images). when this tool is stable, i think it will merge back to docker's namespace.

Reply:
I mean you are using `tonistiigi/binfmt` instead of official image `linuxkit/binfmt` Is there any particular reason for that ?

Reply:
no special reason, just it is on README of buildx Which is widely used by github actions.

https://github.com/docker/buildx#building-multi-platform-images
![image](https://user-images.githubusercontent.com/1667873/97263808-af2ebb80-185e-11eb-8067-c9ffcd30dd90.png)

Reply:
May be the version of qemu.

`tonistiigi/binfmt` with qemu@5.0
`linuxkit/binfmt` with qemu@4.2

Reply:
Ahh I see, thank you @morlay that answers my question.